sale_stock_picking_invoicing/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2020-TODAY Akretion
+# @author Magno Costa <magno.costa@akretion.com.br>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _get_invoiceable_lines(self, final=False):
+        """Return the invoiceable lines for order `self`."""
+        lines = super()._get_invoiceable_lines(final)
+        model = self.env.context.get("active_model")
+        if (
+            self.company_id.sale_invoicing_policy == "stock_picking"
+            and model != "stock.picking"
+        ):
+            new_lines = lines.filtered(lambda ln: ln.product_id.type != "product")
+            if new_lines:
+                # Case lines with Product Type 'service'
+                lines = new_lines
+            else:
+                # Case only Products Type 'product'
+                raise UserError(
```

Comment:
Could you just continue and don't add the product?

How we gonna deal with this example?

![image](https://github.com/OCA/account-invoicing/assets/1975978/2cb22704-1d40-43f9-bfd7-7fe9ba20bdd0)

![image](https://github.com/OCA/account-invoicing/assets/1975978/4dded996-31cf-4b9f-8328-ad2b62857512)

Reply:
Thanks @mileo for review, the UserError just appear when there are only Lines with Product Type 'product' if there are any Service line the Invoice will be create, it's a way to inform the user about the "Sale Invoicing Policy" ( similar to the message when Product must be Delivered to be able Create the Invoice ), in the case of Services the 'product' must be defined as Type 'service'

![image](https://github.com/OCA/account-invoicing/assets/6341149/92a020dc-6e0e-461e-9b0b-a438a9bb0108)

Demo data has a Sale Order of the case of Product and Service, for now in this case will be create two Invoices, one with the Service created from Sale Order and the other from Picking, 

![image](https://github.com/OCA/account-invoicing/assets/6341149/c3081ea9-6011-4318-96b8-3eae92718ade)

![image](https://github.com/OCA/account-invoicing/assets/6341149/ad8ae34a-6f87-4ac5-8f66-f3ba533d4f30)

Service Invoice

![image](https://github.com/OCA/account-invoicing/assets/6341149/a0cc2160-a24d-4257-add0-02313e5f6e80)

Try to create Invoice from Sale Order, there lines but the policy chosen are Stock Picking

![image](https://github.com/OCA/account-invoicing/assets/6341149/568382ce-d8ac-4588-8855-2ffc83695e07)

Create from Stock Picking

![image](https://github.com/OCA/account-invoicing/assets/6341149/368ab9c0-0ce0-4a4f-a7bc-e240d1fab789)

![image](https://github.com/OCA/account-invoicing/assets/6341149/16f3b8f4-80da-4aa8-9179-271dff70b5a2)

![image](https://github.com/OCA/account-invoicing/assets/6341149/76bc1d7c-ca12-44d2-a80a-6bc4158c9b7e)

![image](https://github.com/OCA/account-invoicing/assets/6341149/16e941fd-09c8-4fa9-850b-ddbf763613e7)

This is the expected behavior in brazilian localization, but I think this can be analise, debate and if necessary change, for example:

- If Policy are Stock Picking create only one Invoice from Picking( get Services Lines in the same way of 'Down Payments' )
- If there are Product and Service lines ignore the "Sale Invoicing Policy" and allow Create from Sale Order
- Other ideas